### PR TITLE
Build/Test/Push AD container images based on nightly Samba RPMs

### PR DESCRIPTION
**Pull request by anoopcs9**
This is something I had running under my fork for sometime and now rebased on top of https://github.com/samba-in-kubernetes/samba-container/pull/66.

@phlogistonjohn let me know your thoughts..

### CODE REVIEW

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
First patch looks fine (changing the rpm repo location).

The next patch feels like it's doing to much in one patch and while I get the gist I have a difficult time following it.  Can I suggest you break it up into around 3 patches: one for the build changes, one for the test changes and one for the CI/infra changes? Thanks.

**Comment by anoopcs9** [comment]
> First patch looks fine (changing the rpm repo location).
> 
> The next patch feels like it's doing to much in one patch and while I get the gist I have a difficult time following it. Can I suggest you break it up into around 3 patches: one for the build changes, one for the test changes and one for the CI/infra changes? Thanks.

Done.

File: images/ad-server/Containerfile [hunk 1]
```diff
@@ -1,7 +1,10 @@
 FROM registry.fedoraproject.org/fedora:34
 
-RUN dnf install -y samba-dc procps-ng /usr/bin/smbclient
-RUN rm -f /etc/samba/smb.conf
+ARG INSTALL_PACKAGES_FROM=default
```
Comment (phlogistonjohn) [inline images/ad-server/Containerfile:3]: Even if we don't do much with it, we should probably also have a "SAMBA_SPECIFICS" arg here like we do in the server image as a "breadcrumb"

File: tests/test-deploy-ad-member.sh [hunk 2]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env bash
 
 SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
+IMG_TAG=${IMG_TAG:-"latest"}
 
 source "${SCRIPT_DIR}/common.sh"
 
 echo "Creating ad member pod..."
-ERROR_MSG=$(kubectl create -f "${MEMBER_POD_YAML}" 2>&1 1>/dev/null)
+ERROR_MSG=$(IMG_TAG=${IMG_TAG} envsubst < "${MEMBER_POD_YAML}" | kubectl create -f - 2>&1 1>/dev/null)
```
Comment (phlogistonjohn) [inline tests/test-deploy-ad-member.sh:9]: Not my ideal form of templating, but I think its good enough for now.

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
Only thing I needs doing is the SAMBA_SPECIFICS remark. Everything else is just commentary. Once that's done we can probably merge.

File: images/ad-server/Containerfile (hunk 1 above)
Comment (anoopcs9) [reply images/ad-server/Containerfile:3]: Fine.

**Review by phlogistonjohn — APPROVED** [review]

**Pull request merged**